snowflake/src/test/java/org/apache/iceberg/snowflake/NamespaceHelpersTest.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.snowflake;
+
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+
+public class NamespaceHelpersTest {
+  @Test
+  public void testToSnowflakeIdentifierRoot() {
+    Assertions.assertThat(NamespaceHelpers.toSnowflakeIdentifier(Namespace.empty()))
+        .isEqualTo(SnowflakeIdentifier.ofRoot());
+  }
+
+  @Test
+  public void testToSnowflakeIdentifierDatabase() {
+    Assertions.assertThat(NamespaceHelpers.toSnowflakeIdentifier(Namespace.of("DB1")))
+        .isEqualTo(SnowflakeIdentifier.ofDatabase("DB1"));
+  }
+
+  @Test
+  public void testToSnowflakeIdentifierSchema() {
+    Assertions.assertThat(NamespaceHelpers.toSnowflakeIdentifier(Namespace.of("DB1", "SCHEMA1")))
+        .isEqualTo(SnowflakeIdentifier.ofSchema("DB1", "SCHEMA1"));
+  }
+
+  @Test
+  public void testToSnowflakeIdentifierMaxNamespaceLevel() {
+    Assertions.assertThatExceptionOfType(IllegalArgumentException.class)
+        .isThrownBy(
+            () ->
+                NamespaceHelpers.toSnowflakeIdentifier(
+                    Namespace.of("DB1", "SCHEMA1", "THIRD_NS_LVL")))
+        .withMessageContaining("max namespace level");
+  }
+
+  @Test
+  public void testToSnowflakeIdentifierTable() {
+    Assertions.assertThat(
+            NamespaceHelpers.toSnowflakeIdentifier(TableIdentifier.of("DB1", "SCHEMA1", "TABLE1")))
+        .isEqualTo(SnowflakeIdentifier.ofTable("DB1", "SCHEMA1", "TABLE1"));
+  }
+
+  @Test
+  public void testToSnowflakeIdentifierTableBadNamespace() {
+    Assertions.assertThatExceptionOfType(IllegalArgumentException.class)
+        .isThrownBy(
+            () ->
+                NamespaceHelpers.toSnowflakeIdentifier(
+                    TableIdentifier.of(Namespace.of("DB1_WITHOUT_SCHEMA"), "TABLE1")))
+        .withMessageContaining("must be at the SCHEMA level");
+  }
+
+  @Test
+  public void testToIcebergNamespaceRoot() {
+    Assertions.assertThat(NamespaceHelpers.toIcebergNamespace(SnowflakeIdentifier.ofRoot()))
```

Comment:
nit: I think it's ok to combine `toSnowflakeIdentifier` and `toIcebergNamespace` into a single test method rather than having two separate ones. That way you directly test the round-trip of each in a single method.

The same applies for all the other test methods. So instead of having `testToSnowflakeIdentifierX` & `testToIcebergX` I would just have one named `testX` or something similar

Reply:
Good point about round trips; I do normally have a tendency to write single large test cases just for convenience, so I was trying to do better about following https://www.oreilly.com/library/view/java-extreme-programming/0596003870/ch04s05.html :)

But I agree the right granularity is flexible depending on the scenarios. I went ahead and made all the "success" test cases be combined round-trip tests instead, and kept "failure" cases separate.